src/printer.js

```diff
@@ -626,7 +628,7 @@ function genericPrintNoParens(path, options, print) {
           concat([
             leftBrace,
             indent(
-              options.tabWidth,
+              options.tabWidth * (parentIsUnionTypeAnnotation ? 2 : 1),
```

Comment:
Actually, indentations are only applied when emitting newlines so you need to include the newline:

```js
indent(parentIsUnionTypeAnnotation ? 3 : 0,
       concat([options.bracketSpacing ? line : softline, rightBrace]))
```

Haven't actually testing this code